packages/bitcore-node/src/providers/chain-state/internal/internal.ts

```diff
@@ -201,6 +201,21 @@ export class InternalStateProvider implements CSP.IChainStateService {
     }
   }
 
+  async getRawTransaction(params: CSP.StreamTransactionParams) {
+    let { chain, network, txId } = params;
+    if (typeof txId !== 'string' || !chain || !network) {
+      throw 'Missing required param';
+    }
+    network = network.toLowerCase();
+    let rpcTx = {} as  RPCTransaction;
+    
+    this.getRPC(chain, network).getTransaction(txId, (_, tx: RPCTransaction) => {
+      rpcTx = tx;
+    })
+    
+    return { hex: rpcTx.hex };
```

Comment:
This is going to return before the RPC request callback is invoked (not awaited). You should probably just wrap the whole RPC call in a promise.

```
return new Promise((resolve, reject) => {
      this.getRPC(chain, network).getTransaction(txId, (err: any, result: any) => {
        if (err) {
          reject(err);
        } else {
          resolve({ hex: rpcTx.hex });
        }
      });
```

Reply:
I see, ok I changed it here fe2bf110fd605b943f3d03f0563c68eb1e8bffea